File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOutlineVariantLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOutlineVariantLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl46
            AppTheme.STANDARD -> R.color.cstl46
            AppTheme.BLUE -> R.color.cbl46
            AppTheme.GREEN -> R.color.cgl46
            AppTheme.RED -> R.color.crl46
            AppTheme.NOTHING -> R.color.cnol46
        }
    }
}

File: lib/layer2_core/light/e_outline_variant_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOutlineVariantLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl46;
      case 'standard': return cstl46;
      case 'blue': return cbl46;
      case 'green': return cgl46;
      case 'red': return crl46;
      case 'nothing': return cnol46;
      default: return cstl46;
    }
  }
}
